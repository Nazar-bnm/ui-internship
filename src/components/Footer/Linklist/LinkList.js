import React from 'react';
import PropTypes from 'prop-types';

import './LinkList.scss';

const Linklist = ({ links }) => {
  const { heading, data } = links;

  return (
    <nav>
      <h3 className="linklist__title">{heading}</h3>
      {data.map(({ link, title }) => (
        <a href={link} key={title}>
          <h4 className="linklist__text">{title}</h4>
        </a>
      ))}
    </nav>
  );
};

Linklist.propTypes = {
  links: PropTypes.object.isRequired
};

export default Linklist;
