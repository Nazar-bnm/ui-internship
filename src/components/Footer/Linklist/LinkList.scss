@import '../../../scss/global_styles.scss';

.linklist {
  &__title {
    @include footer-title;
  }

  &__text{
    @include footer-text;
  }
}

