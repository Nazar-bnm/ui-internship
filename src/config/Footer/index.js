export const linkListItems = {
  guide: {
    heading: 'Shopping guide',
    data: [
      { title: 'About G-Star Raw', link: '#' },
      { title: 'Corporate Responsibility', link: '#' },
      { title: 'Press Room', link: '#' },
      { title: 'Careers', link: '#' },
      { title: 'G-Star Retailers', link: '#' },
      { title: 'Open a G-Star Store', link: '#' },
    ],
  },
  help: {
    heading: 'Help & Info',
    data: [
      { title: 'FAQ', link: '#' },
      { title: 'Contact', link: '#' },
      { title: 'Privacy Policy', link: '#' },
      { title: 'Terms & Conditions', link: '#' },
      { title: 'Disclaimer', link: '#' },
    ],
  },
};

export const icons = [
  { name: 'twitter', link: 'https://twitter.com/griddynamics' },
  { name: 'mail', link: 'mailto:khrystynavaskovych@gmail.com' },
  { name: 'facebook', link: 'https://www.facebook.com/griddynamics' },
  { name: 'linkedin', link: 'https://www.linkedin.com/company/grid-dynamics/' },
  { name: 'instagram', link: 'https://www.instagram.com/griddynamics_ua/' },
  {
    name: 'google plus g',
    link: 'https://plus.google.com/115302417170674279390',
  },
];

export const paymentIcons = [
  { name: 'cc paypal', link: '#' },
  { name: 'cc visa', link: '#' },
  { name: 'cc mastercard', link: '#' },
];

export const linkListItemsTablet = [
  { title: 'About G-Star Raw', link: '#' },
  { title: 'G-Star Retailers', link: '#' },
  { title: 'FAQ', link: '#' },
  { title: 'Contact', link: '#' },
  { title: 'Privacy Policy', link: '#' },
  { title: 'Terms & Conditions', link: '#' },
];
