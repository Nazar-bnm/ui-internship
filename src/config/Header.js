const navConfig = [
  /* links should be added after corresponsive
    pages are implemented */
  { value: 'women', label: 'women', link: '/404' },
  { value: 'men', label: 'men', link: '/404' },
  { value: 'kids', label: 'kids', link: '/404' },
  { value: 'accs', label: 'accessories', link: '/404' },
  { value: 'new', label: 'whats new', link: '/404' },
  { value: 'brands', label: 'brands', link: '/404' },
  { value: 'sale', label: 'sale', link: '/404' },
  { value: 'blog', label: 'blog', link: '/404' },
];

export default navConfig;
