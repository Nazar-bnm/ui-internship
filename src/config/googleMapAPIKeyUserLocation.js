export const googleMapAPIKeyUserLocation = 'AIzaSyBAjVWlquT_qSKufJyzldhchRMOw46WA1I';