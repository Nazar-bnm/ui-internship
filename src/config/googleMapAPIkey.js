export const googleAPIKey = 'AIzaSyAyesbQMyKVVbBgKVi2g6VX7mop2z96jB';
