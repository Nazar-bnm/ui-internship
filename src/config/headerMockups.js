export const navConfig = [
  { value: 'women', label: 'women', link: '/women' },
  { value: 'men', label: 'men', link: '/not_found' },
  { value: 'kids', label: 'kids', link: '/not_found' },
  { value: 'accs', label: 'accessories', link: '/not_found' },
  { value: 'new', label: 'what\'s new', link: '/not_found' },
  { value: 'brands', label: 'brands', link: '/not_found' },
  { value: 'sale', label: 'sale', link: '/not_found' },
  { value: 'blog', label: 'blog', link: '/not_found' },
];

export const options = {
  currency: [
    { value: 'USD', label: 'USD' },
    { value: 'EURO', label: 'EURO' },
    { value: 'UAH', label: 'UAH' },
  ],
  language: [
    { value: 'EN', label: 'English' },
    { value: 'RU', label: 'Russian' },
    { value: 'UA', label: 'Ukrainian' },
  ],
  company: [
    { value: 'Co', label: 'Company' },
    { value: 'Co2', label: 'Company 2' },
    { value: 'Co3', label: 'Company 3' },
  ],
};

export const burgerMenuItemsList = [
  { itemName: 'woman', url: 'woman' },
  { itemName: 'man', url: 'man' },
  { itemName: 'kids', url: 'kids' },
  { itemName: 'accessories', url: '404' },
  { itemName: 'what\'s new', url: '404' },
  { itemName: 'brands', url: '404' },
  { itemName: 'sale', url: '404' },
  { itemName: 'blog', url: '404' },
];
