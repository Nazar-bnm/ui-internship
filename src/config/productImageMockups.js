export const images = [
  {
    // eslint-disable-next-line max-len
    src: 'https://www.nationalgeographic.com/content/dam/animals/thumbs/rights-exempt/mammals/r/raccoon_thumb.ngsversion.1485815402351.adapt.1900.1.JPG',
  },
  {
    // eslint-disable-next-line max-len
    src: 'https://www.nydailynews.com/resizer/57h4sk1Mr8eFVp6LV7LlJ3zBasI=/800x600/top/arc-anglerfish-arc2-prod-tronc.s3.amazonaws.com/public/CVQJPCXX35C5NFMJMPHHC6QKHQ.jpg',
  },
  {
    src: 'https://cdn.mos.cms.futurecdn.net/xCknw5UTKR8rM97eZE7reF-320-80.jpg',
  },
  {
    // eslint-disable-next-line max-len
    src: 'https://curiodyssey.org/wp-content/uploads/bb-plugin/cache/xMammals-Raccoon-square.jpg.pagespeed.ic.W5-0FMTZBa.jpg',
  },
];
