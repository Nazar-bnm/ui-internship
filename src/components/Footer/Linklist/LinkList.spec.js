import React from 'react';
import { shallow } from 'enzyme';
import LinkList from './LinkList';
import config from '../../../../config';

describe('<LinkList />', () => {
  const { linkListItems } = config;
  const { guide } = linkListItems;

  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      links: guide,
    };

    wrapper = shallow(
      <LinkList {...props} />
    );
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
