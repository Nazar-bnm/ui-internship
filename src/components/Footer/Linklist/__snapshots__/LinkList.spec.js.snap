// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkList /> should match snapshot 1`] = `
<nav>
  <h3
    className="linklist__title"
  >
    Shopping guide
  </h3>
  <a
    href="#"
    key="About G-Star Raw"
  >
    <h4
      className="linklist__text"
    >
      About G-Star Raw
    </h4>
  </a>
  <a
    href="#"
    key="Corporate Responsibility"
  >
    <h4
      className="linklist__text"
    >
      Corporate Responsibility
    </h4>
  </a>
  <a
    href="#"
    key="Press Room"
  >
    <h4
      className="linklist__text"
    >
      Press Room
    </h4>
  </a>
  <a
    href="#"
    key="Careers"
  >
    <h4
      className="linklist__text"
    >
      Careers
    </h4>
  </a>
  <a
    href="#"
    key="G-Star Retailers"
  >
    <h4
      className="linklist__text"
    >
      G-Star Retailers
    </h4>
  </a>
  <a
    href="#"
    key="Open a G-Star Store"
  >
    <h4
      className="linklist__text"
    >
      Open a G-Star Store
    </h4>
  </a>
</nav>
`;
